xds/internal/server/listener_wrapper_test.go

```diff
@@ -233,204 +305,72 @@ func (fc *fakeConn) Close() error {
-		}}, nil)
-
-	// This should start a watch on xds client for rds name "route-1".
-	routeName, err := xdsC.WaitForWatchRouteConfig(ctx)
-	if err != nil {
-		t.Fatalf("error when waiting for a watch on a Route resource: %v", err)
-	}
-	if routeName != "route-1" {
-		t.Fatalf("listenerWrapper registered a lds watch on %s, want %s", routeName, "route-1")
-	}
+	waitForResourceNames(ctx, t, ldsResourceNamesCh, []string{lisResourceName})
 
-	// This shouldn't invoke good update channel, as has not received rds updates yet.
-	timer := time.NewTimer(defaultTestShortTimeout)
-	select {
-	case <-timer.C:
-		timer.Stop()
-	case <-readyCh:
-		t.Fatalf("ready channel written to without rds configuration specified")
+	// Configure the management server with a listener resource.
+	resources := e2e.UpdateOptions{
+		NodeID:         nodeID,
+		Listeners:      []*v3listenerpb.Listener{e2e.DefaultServerListener(fakeListenerHost, fakeListenerPort, e2e.SecurityLevelNone, route1)},
+		SkipValidation: true,
 	}
-
-	// Invoke rds callback for the started rds watch. This valid rds callback
-	// should trigger the listener wrapper to fire GoodUpdate, as it has
-	// received both it's LDS Configuration and also RDS Configuration,
-	// specified in LDS Configuration.
-	xdsC.InvokeWatchRouteConfigCallback("route-1", xdsresource.RouteConfigUpdate{}, nil)
-
-	// All of the xDS updates have completed, so can expect to send a ping on
-	// good update channel.
-	select {
-	case <-ctx.Done():
-		t.Fatalf("timeout waiting for the ready channel to be written to after receipt of a good rds update")
-	case <-readyCh:
-	}
-}
-
-func (s) TestListenerWrapper_Accept(t *testing.T) {
-	boCh := testutils.NewChannel()
-	origBackoffFunc := backoffFunc
-	backoffFunc = func(v int) time.Duration {
-		boCh.Send(v)
-		return 0
+	if err := mgmtServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
 	}
-	defer func() { backoffFunc = origBackoffFunc }()
 
-	lw, readyCh, xdsC, lis, cleanup := newListenerWrapper(t)
-	defer cleanup()
-
-	// Push a good update with a filter chain which accepts local connections on
-	// 192.168.0.0/16 subnet and port 80.
-	fcm, err := xdsresource.NewFilterChainManager(listenerWithFilterChains(t))
-	if err != nil {
-		t.Fatalf("xdsclient.NewFilterChainManager() failed with error: %v", err)
-	}
-	xdsC.InvokeWatchListenerCallback(xdsresource.ListenerUpdate{
-		InboundListenerCfg: &xdsresource.InboundListenerConfig{
-			Address:      fakeListenerHost,
-			Port:         strconv.Itoa(fakeListenerPort),
-			FilterChains: fcm,
-		}}, nil)
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-	defer close(lis.acceptCh)
+	// Verify that the listener wrapper becomes ready.
 	select {
 	case <-ctx.Done():
-		t.Fatalf("timeout waiting for the ready channel to be written to after receipt of a good Listener update")
+		t.Fatalf("Timeout waiting for the ready channel to be written to after receipt of a good Listener update")
```

Comment:
It seems like you don't need inline route configuration to trigger ready, just no rds specified in filter chains? Can you test no inline route update/no specified rds name too? I think should become ready too.